client/docker/images.go

```
@@ -29,32 +29,96 @@ import (
 )
 
-func (d Docker) ImagesPull(images []string) error {
+func (d Docker) ImagesCacheToRegistry(images []string) error {
+	version, err := d.cli.ServerVersion(d.ctx)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(version.Version, `sealer`) {
+		return d.ImagesPull(images)
+	}
+	// need a extra push step
+	if err := d.ImagesPull(images); err != nil {
+		return err
+	}
+	return d.ImagesPush(images)
+}
+func (d Docker) ImagesPush(images []string) error {
 	for _, image := range utils.RemoveDuplicate(images) {
 		if image == "" {
 			continue
 		}
 		if strings.HasPrefix(image, "#") {
 			continue
 		}
-		if err := d.ImagePull(strings.TrimSpace(image)); err != nil {
-			return fmt.Errorf("image %s pull failed: %v", image, err)
+		if err := d.ImagePush(strings.TrimSpace(image)); err != nil {
+			return fmt.Errorf("image %s push failed: %v", image, err)
 		}
 	}
 	return nil
 }
-
-func (d Docker) ImagesPullByImageListFile(fileName string) error {
-	data, err := utils.ReadLines(fileName)
+func (d Docker) ImagePush(image string) error {
+	var (
+		err          error
+		out          io.ReadCloser
+		named        reference.Named
+		registryName = `sea.hub:5000`
+		TagImageName string
+	)
+	named, err = GetCanonicalImageName(image)
 	if err != nil {
-		logger.Error(fmt.Sprintf("Read image list failed: %v", err))
+		return fmt.Errorf("failed to parse canonical image name %s : %v", image, err)
 	}
-	return d.ImagesPull(data)
-}
+	TagImageName = strings.Replace(named.String(), named.String()[0:strings.Index(named.String(), `/`)], registryName, 1)
+	opts := types.ImagePushOptions(GetCanonicalImagePullOptions(named.String()))
+	if opts.RegistryAuth == "" {
+		opts.RegistryAuth = "sea.hub" //must not empty
+	}
+	err = d.cli.ImageTag(d.ctx, named.String(), TagImageName)
+	if err != nil {
+		return err
+	}
+	out, err = d.cli.ImagePush(d.ctx, TagImageName, opts)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = out.Close()
```

should not ignore errors